lab/src/main/ii.go

```diff
@@ -1,22 +1,54 @@
 package main
 
-import "os"
-import "fmt"
-import "mapreduce"
+import (
+	"fmt"
+	"os"
+	"mapreduce"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
 
 // The mapping function is called once for each piece of the input.
 // In this framework, the key is the name of the file that is being processed,
 // and the value is the file's contents. The return value should be a slice of
 // key/value pairs, each represented by a mapreduce.KeyValue.
 func mapF(document string, value string) (res []mapreduce.KeyValue) {
 	// Your code here (Part V).
+	rege, _ := regexp.Compile("([a-z]|[A-Z])+")
+
+	words := rege.FindAllString(value, -1)
+
+	for _, word := range words {
+		res = append(res, mapreduce.KeyValue{word, document})
+	}
+	return
 }
 
 // The reduce function is called once for each key generated by Map, with a
 // list of that key's string value (merged across all inputs). The return value
 // should be a single output value for that key.
 func reduceF(key string, values []string) string {
 	// Your code here (Part V).
+	sort.Strings(values)
+	distinctDocuments := removeDuplicate(values)
+
+	documents := strings.Join(distinctDocuments, ",")
+	value := strconv.Itoa(len(distinctDocuments)) + " " + documents
+	return value
+}
+
+func removeDuplicate(values []string) []string {
+	encountered := make(map[string]bool)
+	result := []string{}
+	for _, value := range values {
+		if _, ok := encountered[value]; !ok {
+			encountered[value] = true
```

Comment:
[nit] We don't have to update the value (i.e. the bool) of the map, since we're not using it anywhere, and we only check if the key (i.e. the string "value") is in the map by "ok".